src/components/element-editor/validation/validation.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+import { MultiEntry } from "src/components/common";
+import { ValidationChild } from "./validation-child";
+import { EElementType, useBuilder } from "src/context-providers";
+import { ELEMENT_VALIDATION_TYPES } from "src/data";
+import { SimpleIdGenerator } from "src/util/simple-id-generator";
+
+export interface IEntryProps {
+    type?: string;
+    rule?: string;
+    errorMessage?: string;
+}
+
+export const Validation = () => {
+    // =========================================================================
+    // CONST, STATES, REFS
+    // =========================================================================
+    const { focusedElement } = useBuilder();
+    const element = focusedElement.element;
+    const [numberOfChildEntry, setNumberOfChildEntry] = useState(0);
+    const [childEntryValues, setChildEntryValues] = useState<IEntryProps[]>([]);
+    const [childKey, setChildKey] = useState([]);
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    function getOptionsByType(elementType: EElementType) {
+        switch (elementType) {
+            case EElementType.EMAIL:
+                return ELEMENT_VALIDATION_TYPES["Text field"][
+                    EElementType.EMAIL
+                ];
+            default:
+                return ["Select", "Type 1"];
+        }
+    }
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleAddButtonClick = () => {
+        const newKey = SimpleIdGenerator.generate();
+        setChildKey((prevValue) => [...prevValue, newKey]);
+
+        setNumberOfChildEntry((num) => num + 1);
+    };
+
+    const handleBinIconClick = (type: string) => {
+        if (childEntryValues.length > 0) {
+            const updatedValues = childEntryValues.filter(
+                (child) => child?.type !== type
+            );
+            const childToRemove = childEntryValues.find(
+                (child) => child?.type === type
+            );
+            const updatedKeys = childKey.filter(
+                (_, index) => index !== childEntryValues.indexOf(childToRemove)
+            );
+
+            setChildEntryValues(updatedValues);
+            setChildKey(updatedKeys);
+            setNumberOfChildEntry((num) => num - 1);
+        }
+    };
```

Comment:
- rename as `handleDelete`
- instead of type, pass index as argument, you can remove the childEntryValue by the index